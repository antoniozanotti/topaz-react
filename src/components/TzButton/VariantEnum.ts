export enum VariantEnum {
  accent = 'accent',
  primary = 'primary',
  secondary = 'secondary',
  negative = 'negative',
  dark = 'dark',
  light = 'light',
}
