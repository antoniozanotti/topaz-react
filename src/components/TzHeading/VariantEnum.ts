export enum VariantEnum {
  'accent' = 'accent',
  'accent-1' = 'accent-1',
  'accent-2' = 'accent-2',
  'primary' = 'primary',
  'primary-1' = 'primary-1',
  'primary-2' = 'primary-2',
  'secondary' = 'secondary',
  'secondary-1' = 'secondary-1',
  'secondary-2' = 'secondary-2',
  'negative' = 'negative',
  'negative-1' = 'negative-1',
  'negative-2' = 'negative-2',
  'dark' = 'dark',
  'dark-1' = 'dark-1',
  'dark-2' = 'dark-2',
  'light' = 'light',
  'light-1' = 'light-1',
  'light-2' = 'light-2',
}
