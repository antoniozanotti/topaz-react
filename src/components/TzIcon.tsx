import React from 'react';
import * as heroIcons from '@heroicons/react/24/outline';

export interface TzIconProps {
  iconName: keyof typeof heroIcons;
  variant: 'accent' | 'primary' | 'secondary' | 'negative' | 'dark' | 'light';
  className?: string;
}

export const TzIcon = ({
  iconName,
  variant = 'accent',
  className = ''
}: TzIconProps) => {

  // size
  let sizeClasses = 'w-[12px] h-[12px] sm:w-[16px] sm:h-[16px] lg:w-[20px] lg:h-[20px]';

  // variants
  let variantClasses = '';
  if (variant == 'accent') {
    variantClasses = 'text-accent dark:text-dark-accent';
  } else if (variant == 'primary') {
    variantClasses = 'text-primary dark:text-dark-primary';
  } else if (variant == 'secondary') {
    variantClasses = 'text-secondary dark:text-dark-secondary';
  } else if (variant == 'negative') {
    variantClasses = 'text-negative dark:text-dark-negative';
  } else if (variant == 'dark') {
    variantClasses = 'text-dark dark:text-dark-dark';
  } else if (variant == 'light') {
    variantClasses = 'text-light dark:text-dark-light';
  }

  // icon
  let iconComponentName = iconName ? heroIcons[iconName] : '';

  return (
    React.createElement(iconComponentName, {
      className: `${sizeClasses} ${variantClasses} ${className}`,
    })
  );
};
