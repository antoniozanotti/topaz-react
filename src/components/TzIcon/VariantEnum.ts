export enum VariantEnum {
  'accent' = 'accent',
  'inverse-accent' = 'inverse-accent',
  'primary' = 'primary',
  'inverse-primary' = 'inverse-primary',
  'secondary' = 'secondary',
  'inverse-secondary' = 'inverse-secondary',
  'negative' = 'negative',
  'inverse-negative' = 'inverse-negative',
  'dark' = 'dark',
  'inverse-dark' = 'inverse-dark',
  'light' = 'light',
  'inverse-light' = 'inverse-light',
}
