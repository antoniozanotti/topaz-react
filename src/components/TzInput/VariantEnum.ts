export enum VariantEnum {
  ACCENT = 'accent',
  PRIMARY = 'primary',
  SECONDARY = 'secondary',
  NEGATIVE = 'negative',
  DARK = 'dark',
  LIGHT = 'light',
}