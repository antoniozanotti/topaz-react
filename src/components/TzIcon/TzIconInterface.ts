import * as heroIcons from '@heroicons/react/24/outline';
import { VariantEnum } from './VariantEnum';

export interface TzIconInterface {
  iconName: keyof typeof heroIcons;
  variant?: VariantEnum;
  className?: string;
}