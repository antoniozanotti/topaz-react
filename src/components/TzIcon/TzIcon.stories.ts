import type { Meta, StoryObj } from "@storybook/react";
import { TzIcon } from "./TzIcon";

/**
 * The TzIcon component based on [heroicons](https://heroicons.com/).
 */
const meta: Meta<typeof TzIcon> = {
  component: TzIcon,
  tags: ["autodocs"],
  args: {
    iconName: "ArrowRightIcon",
    variant: "light",
  },
};

export default meta;
type Story = StoryObj<typeof meta>;

export const arrowRightIcon: Story = {
  name: "ArrowRightIcon, light"
};

export const moon: Story = {
  name: "MoonIcon, primary",
  args: {
    iconName: "MoonIcon",
    variant: "primary"
  },
};

export const sun: Story = {
  name: "SunIcon, secondary",
  args: {
    iconName: "SunIcon",
    variant: "secondary"
  },
};

export const heart: Story = {
  name: "HeartIcon, negative",
  args: {
    iconName: "HeartIcon",
    variant: "negative"
  },
};

export const SmallMobile: Story = {
  name: "Small Mobile",
  parameters: {
    viewport: {
      defaultViewport: "mobile1",
    },
  }
};

export const LargeMobile: Story = {
  name: "Large Mobile",
  parameters: {
    viewport: {
      defaultViewport: "mobile2",
    },
  }
};

export const Tablet: Story = {
  name: "Tablet",
  parameters: {
    viewport: {
      defaultViewport: "tablet",
    },
  }
};

export const Desktop: Story = {
  name: "Desktop"
};

