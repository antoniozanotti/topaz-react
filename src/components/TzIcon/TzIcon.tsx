import React from 'react';
import * as heroIcons from '@heroicons/react/24/outline';
import { TzIconInterface } from './TzIconInterface';
import { VariantEnum } from './VariantEnum';

export function TzIcon({
  iconName,
  variant = VariantEnum.accent,
  className = '',
}: TzIconInterface) {
  const sizeStyles = 'w-[16px] h-[16px] lg:w-[20px] lg:h-[20px]';
  const variantStyles = useVariantStyles(variant);

  return React.createElement(heroIcons[iconName], {
    className: `${sizeStyles} ${variantStyles} ${className}`,
    title: iconName
  });
}

function useVariantStyles(variant: keyof typeof VariantEnum){
  return {
    [VariantEnum['accent']]: 'text-accent dark:text-dark-accent',
    [VariantEnum['inverse-accent']]: 'text-inverse-accent dark:text-inverse-dark-accent',
    [VariantEnum['primary']]: 'text-primary dark:text-dark-primary',
    [VariantEnum['inverse-primary']]: 'text-inverse-primary dark:text-inverse-dark-primary',
    [VariantEnum['secondary']]: 'text-secondary dark:text-dark-secondary',
    [VariantEnum['inverse-secondary']]: 'text-inverse-secondary dark:text-inverse-dark-secondary',
    [VariantEnum['negative']]: 'text-negative dark:text-dark-negative',
    [VariantEnum['inverse-negative']]: 'text-inverse-negative dark:text-inverse-dark-negative',
    [VariantEnum['dark']]: 'text-dark dark:text-dark-dark',
    [VariantEnum['inverse-dark']]: 'text-inverse-dark dark:text-inverse-dark-dark',
    [VariantEnum['light']]: 'text-light dark:text-dark-light',
    [VariantEnum['inverse-light']]: 'text-inverse-light dark:text-inverse-dark-light',
  }[variant];
}