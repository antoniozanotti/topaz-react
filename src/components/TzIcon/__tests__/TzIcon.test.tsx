import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { TzIcon } from '../TzIcon';
import * as heroIcons from '@heroicons/react/24/outline';
import { VariantEnum } from '../VariantEnum';

const basicTestIcon = (
  iconName: keyof typeof heroIcons,
  variant: keyof typeof VariantEnum
) => {
  const { container } = render(<TzIcon iconName={iconName} variant={variant} />);

  const icon = screen.getByTitle(iconName);
  expect(icon).toBeInstanceOf(SVGElement);
  expect(icon).toBeInTheDocument();
  expect(container).toMatchSnapshot();
  
  return icon;
};

describe('TzIcon Component', () => {
  it('should render ArrowRightIcon, light', () => {
    basicTestIcon("ArrowRightIcon", "light");
  });

  it('should render MoonIcon, primary', () => {
    basicTestIcon("MoonIcon", "primary");
  });

  it('should render SunIcon, secondary', () => {
    basicTestIcon("SunIcon", "secondary");
  });

  it('should render HeartIcon, negative', () => {
    basicTestIcon("HeartIcon", "negative");
  });
});
