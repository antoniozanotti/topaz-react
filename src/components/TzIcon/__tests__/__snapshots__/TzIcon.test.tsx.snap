// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TzIcon Component should render ArrowRightIcon, light 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="w-[16px] h-[16px] lg:w-[20px] lg:h-[20px] text-light dark:text-dark-light "
    data-slot="icon"
    fill="none"
    stroke="currentColor"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      ArrowRightIcon
    </title>
    <path
      d="M13.5 4.5 21 12m0 0-7.5 7.5M21 12H3"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
`;

exports[`TzIcon Component should render HeartIcon, negative 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="w-[16px] h-[16px] lg:w-[20px] lg:h-[20px] text-negative dark:text-dark-negative "
    data-slot="icon"
    fill="none"
    stroke="currentColor"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      HeartIcon
    </title>
    <path
      d="M21 8.25c0-2.485-2.099-4.5-4.688-4.5-1.935 0-3.597 1.126-4.312 2.733-.715-1.607-2.377-2.733-4.313-2.733C5.1 3.75 3 5.765 3 8.25c0 7.22 9 12 9 12s9-4.78 9-12Z"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
`;

exports[`TzIcon Component should render MoonIcon, primary 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="w-[16px] h-[16px] lg:w-[20px] lg:h-[20px] text-primary dark:text-dark-primary "
    data-slot="icon"
    fill="none"
    stroke="currentColor"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      MoonIcon
    </title>
    <path
      d="M21.752 15.002A9.72 9.72 0 0 1 18 15.75c-5.385 0-9.75-4.365-9.75-9.75 0-1.33.266-2.597.748-3.752A9.753 9.753 0 0 0 3 11.25C3 16.635 7.365 21 12.75 21a9.753 9.753 0 0 0 9.002-5.998Z"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
`;

exports[`TzIcon Component should render SunIcon, secondary 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="w-[16px] h-[16px] lg:w-[20px] lg:h-[20px] text-secondary dark:text-dark-secondary "
    data-slot="icon"
    fill="none"
    stroke="currentColor"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      SunIcon
    </title>
    <path
      d="M12 3v2.25m6.364.386-1.591 1.591M21 12h-2.25m-.386 6.364-1.591-1.591M12 18.75V21m-4.773-4.227-1.591 1.591M5.25 12H3m4.227-4.773L5.636 5.636M15.75 12a3.75 3.75 0 1 1-7.5 0 3.75 3.75 0 0 1 7.5 0Z"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
`;
